<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
 <HEAD>
   <TITLE> [Twisted-web] Exceptions in twisted.web.client
   </TITLE>
   <LINK REL="Index" HREF="index.html" >
   <LINK REL="made" HREF="mailto:twisted-web%40twistedmatrix.com?Subject=%5BTwisted-web%5D%20Exceptions%20in%20twisted.web.client&In-Reply-To=">
   <META NAME="robots" CONTENT="index,nofollow">
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="004334.html">
   <LINK REL="Next"  HREF="004339.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[Twisted-web] Exceptions in twisted.web.client</H1>
    <B>Chris Were</B> 
    <A HREF="mailto:twisted-web%40twistedmatrix.com?Subject=%5BTwisted-web%5D%20Exceptions%20in%20twisted.web.client&In-Reply-To="
       TITLE="[Twisted-web] Exceptions in twisted.web.client">chris.were at gmail.com
       </A><BR>
    <I>Tue Oct  6 23:33:11 EDT 2009</I>
    <P><UL>
        <LI>Previous message: <A HREF="004334.html">[Twisted-web] I can't be the only one who prefers trailing slashes	on my URLs...
</A></li>
        <LI>Next message: <A HREF="004339.html">[Twisted-web] Exceptions in twisted.web.client
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#4335">[ date ]</a>
              <a href="thread.html#4335">[ thread ]</a>
              <a href="subject.html#4335">[ subject ]</a>
              <a href="author.html#4335">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE>Hi,
I have an application that fetches many urls, following any redirects. Every
so often I get a variety of unhandled exceptions from within
twisted.web.client (see snippet at the bottom of email).

My code has an addErrback attached to the deferred and a try/except around
the getPage() call, neither of which get invoked.

Are these exceptions that web.client needs to catch internally and then call
self.deferred.errback so that my error handler will receive it?

If not, how can I catch them?

Cheers,
Chris

--- Exception 1 ---

2009-10-06 20:21:16-0700 [HTTPPageGetter,client] Unhandled Error
 Traceback (most recent call last):
  File
&quot;/usr/local/lib/python2.6/site-packages/Twisted-8.2.0-py2.6-linux-x86_64.egg/twisted/python/log.py&quot;,
line 84, in callWithLogger
    return callWithContext({&quot;system&quot;: lp}, func, *args, **kw)
  File
&quot;/usr/local/lib/python2.6/site-packages/Twisted-8.2.0-py2.6-linux-x86_64.egg/twisted/python/log.py&quot;,
line 69, in callWithContext
    return context.call({ILogContext: newCtx}, func, *args, **kw)
  File
&quot;/usr/local/lib/python2.6/site-packages/Twisted-8.2.0-py2.6-linux-x86_64.egg/twisted/python/context.py&quot;,
line 59, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File
&quot;/usr/local/lib/python2.6/site-packages/Twisted-8.2.0-py2.6-linux-x86_64.egg/twisted/python/context.py&quot;,
line 37, in callWithContext
    return func(*args,**kw)
--- &lt;exception caught here&gt; ---
  File
&quot;/usr/local/lib/python2.6/site-packages/Twisted-8.2.0-py2.6-linux-x86_64.egg/twisted/internet/selectreactor.py&quot;,
line 146, in _doReadOrWrite
    why = getattr(selectable, method)()
  File
&quot;/usr/local/lib/python2.6/site-packages/Twisted-8.2.0-py2.6-linux-x86_64.egg/twisted/internet/tcp.py&quot;,
line 463, in doRead
    return self.protocol.dataReceived(data)
  File
&quot;/usr/local/lib/python2.6/site-packages/Twisted-8.2.0-py2.6-linux-x86_64.egg/twisted/protocols/basic.py&quot;,
line 231, in dataReceived
    why = self.lineReceived(line)
  File
&quot;/usr/local/lib/python2.6/site-packages/Twisted-8.2.0-py2.6-linux-x86_64.egg/twisted/web/http.py&quot;,
line 420, in lineReceived
    self.handleEndHeaders()
  File
&quot;/usr/local/lib/python2.6/site-packages/Twisted-8.2.0-py2.6-linux-x86_64.egg/twisted/web/client.py&quot;,
line 74, in handleEndHeaders
    m()
  File
&quot;/usr/local/lib/python2.6/site-packages/Twisted-8.2.0-py2.6-linux-x86_64.egg/twisted/web/client.py&quot;,
line 93, in handleStatus_301
    _parse(url, defaultPort=self.transport.getPeer().port)
  File
&quot;/usr/local/lib/python2.6/site-packages/Twisted-8.2.0-py2.6-linux-x86_64.egg/twisted/web/client.py&quot;,
line 426, in _parse
    port = int(port)
exceptions.ValueError: invalid literal for int() with base 10: ''

--- Exception 2 ---

2009-10-06 20:28:42-0700 [HTTPPageGetter,client] Unhandled Error
Traceback (most recent call last):
  File
&quot;/usr/local/lib/python2.6/site-packages/Twisted-8.2.0-py2.6-linux-x86_64.egg/twisted/python/log.py&quot;,
line 84, in callWithLogger
    return callWithContext({&quot;system&quot;: lp}, func, *args, **kw)
  File
&quot;/usr/local/lib/python2.6/site-packages/Twisted-8.2.0-py2.6-linux-x86_64.egg/twisted/python/log.py&quot;,
line 69, in callWithContext
    return context.call({ILogContext: newCtx}, func, *args, **kw)
  File
&quot;/usr/local/lib/python2.6/site-packages/Twisted-8.2.0-py2.6-linux-x86_64.egg/twisted/python/context.py&quot;,
line 59, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File
&quot;/usr/local/lib/python2.6/site-packages/Twisted-8.2.0-py2.6-linux-x86_64.egg/twisted/python/context.py&quot;,
line 37, in callWithContext
    return func(*args,**kw)
--- &lt;exception caught here&gt; ---
  File
&quot;/usr/local/lib/python2.6/site-packages/Twisted-8.2.0-py2.6-linux-x86_64.egg/twisted/internet/selectreactor.py&quot;,
line 146, in _doReadOrWrite
    why = getattr(selectable, method)()
  File
&quot;/usr/local/lib/python2.6/site-packages/Twisted-8.2.0-py2.6-linux-x86_64.egg/twisted/internet/tcp.py&quot;,
line 463, in doRead
    return self.protocol.dataReceived(data)
  File
&quot;/usr/local/lib/python2.6/site-packages/Twisted-8.2.0-py2.6-linux-x86_64.egg/twisted/protocols/basic.py&quot;,
line 231, in dataReceived
    why = self.lineReceived(line)
  File
&quot;/usr/local/lib/python2.6/site-packages/Twisted-8.2.0-py2.6-linux-x86_64.egg/twisted/web/http.py&quot;,
line 420, in lineReceived
    self.handleEndHeaders()
  File
&quot;/usr/local/lib/python2.6/site-packages/Twisted-8.2.0-py2.6-linux-x86_64.egg/twisted/web/client.py&quot;,
line 74, in handleEndHeaders
    m()
  File
&quot;/usr/local/lib/python2.6/site-packages/Twisted-8.2.0-py2.6-linux-x86_64.egg/twisted/web/client.py&quot;,
line 93, in handleStatus_301
    _parse(url, defaultPort=self.transport.getPeer().port)
  File
&quot;/usr/local/lib/python2.6/site-packages/Twisted-8.2.0-py2.6-linux-x86_64.egg/twisted/web/client.py&quot;,
line 425, in _parse
    host, port = host.split(':')
exceptions.ValueError: too many values to unpack

--- Exception 3 ---

2009-10-06 20:31:11-0700 [HTTPPageGetter,client] Unhandled Error
Traceback (most recent call last):
  File &quot;/usr/local/lib/python2.6/site-packages/twisted/python/log.py&quot;, line
84, in callWithLogger
    return callWithContext({&quot;system&quot;: lp}, func, *args, **kw)
  File &quot;/usr/local/lib/python2.6/site-packages/twisted/python/log.py&quot;, line
69, in callWithContext
    return context.call({ILogContext: newCtx}, func, *args, **kw)
  File &quot;/usr/local/lib/python2.6/site-packages/twisted/python/context.py&quot;,
line 59, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File &quot;/usr/local/lib/python2.6/site-packages/twisted/python/context.py&quot;,
line 37, in callWithContext
    return func(*args,**kw)
 --- &lt;exception caught here&gt; ---
  File
&quot;/usr/local/lib/python2.6/site-packages/twisted/internet/selectreactor.py&quot;,
line 146, in _doReadOrWrite
    why = getattr(selectable, method)()
  File &quot;/usr/local/lib/python2.6/site-packages/twisted/internet/tcp.py&quot;,
line 463, in doRead
    return self.protocol.dataReceived(data)
  File &quot;/usr/local/lib/python2.6/site-packages/twisted/protocols/basic.py&quot;,
line 231, in dataReceived
    why = self.lineReceived(line)
  File &quot;/usr/local/lib/python2.6/site-packages/twisted/web/http.py&quot;, line
404, in lineReceived
    status = l[1]
exceptions.IndexError: list index out of range
-------------- next part --------------
An HTML attachment was scrubbed...
URL: <A HREF="http://twistedmatrix.com/pipermail/twisted-web/attachments/20091007/310aeb7e/attachment.htm">http://twistedmatrix.com/pipermail/twisted-web/attachments/20091007/310aeb7e/attachment.htm</A> 
</PRE>



<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="004334.html">[Twisted-web] I can't be the only one who prefers trailing slashes	on my URLs...
</A></li>
	<LI>Next message: <A HREF="004339.html">[Twisted-web] Exceptions in twisted.web.client
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#4335">[ date ]</a>
              <a href="thread.html#4335">[ thread ]</a>
              <a href="subject.html#4335">[ subject ]</a>
              <a href="author.html#4335">[ author ]</a>
         </LI>
       </UL>

<hr>
<a href="http://twistedmatrix.com/cgi-bin/mailman/listinfo/twisted-web">More information about the Twisted-web
mailing list</a><br>
</body></html>
